### PR TITLE
[PR-23265][Custom app] Create public documentation

### PR DESCRIPTION
**Pull request by wrimer**
## Demo

https://void-docs.talkable.com/email_marketing_and_automation/custom_app.html

## Related Stories

[](https://talkable.atlassian.net/browse/PR-23265)

### CODE REVIEW

File: source/email_marketing_and_automation/custom_app.rst [hunk 1]
```diff
@@ -66,20 +84,273 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+   * 50.17.244.178
+   * 52.204.148.18
+   * 52.22.0.55
+   * 52.6.41.1
+   * 52.7.94.243
+   * 54.164.128.44
+   * 54.208.14.192
+   * 54.86.208.153
+   * 54.86.241.218
+   * 54.91.51.228
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:356]: Please do not duplicate IPs in documentation and link to existing list instead.
Either that or move list to partial / separate page.
Either way there should only be a single place we’re updating.
https://github.com/talkable/talkable-docs/blob/f8e68648654d7fbf807684ae1944592aa169f38d/source/web_hooks.rst#whitelisting-talkable-ips

File: source/email_marketing_and_automation/custom_app.rst [hunk 2]
```diff
@@ -10,6 +10,24 @@ Custom App
 Custom App allows you to send Talkable data to a desired destination such as your site, ESP, CDP.
 Talkable will send a request with data to the Custom App URL for each customer's event specified in the Custom App settings.
 
+Set Up
+------
+
+1. Navigate into the App store:
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:16]:
```suggestion
1. Navigate to the App store:
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 3]
```diff
@@ -10,6 +10,24 @@ Custom App
...
+
+  .. image:: /_static/img/email_marketing_and_automation/app_store_step_1.png
+    :alt: App store step 1
+
+  .. image:: /_static/img/email_marketing_and_automation/app_store_step_2.png
+    :alt: App store step 2
+
+2. Choose an `Custom app` and clicking “Install”
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:24]:
```suggestion
2. Choose a `Custom app` and clicking “Install”
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 4]
```diff
@@ -66,20 +84,273 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+    );
+
+    return crypto.timingSafeEqual(
+      Buffer.from(calculatedSignature, 'base64'),
+      Buffer.from(headerSignature, 'base64')
+    );
+  }
+
+Available actions
+-----------------
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:106]:
```suggestion
-----------------

```

File: source/email_marketing_and_automation/custom_app.rst [hunk 5]
```diff
@@ -66,20 +84,273 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+Each action allows custom attributes to be included. You can see allowed interpolation variables by clicking `Show available variables` button:
+
+.. image:: /_static/img/email_marketing_and_automation/variables.png
+  :alt: Send same payload
+
+Referral Create
+^^^^^^^^^^^^^^^
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:113]: Please use syntax from the README for headers and subheaders:
https://github.com/talkable/talkable-docs?tab=readme-ov-file#cross-referencing

Please also leave an empty line after all headers.

**Review by zhuravel — CHANGES_REQUESTED** [review]

File: source/email_marketing_and_automation/custom_app.rst [hunk 6]
```diff
@@ -10,6 +10,24 @@ Custom App
...
+Set Up
+------
+
+1. Navigate to the App store:
+
+  .. image:: /_static/img/email_marketing_and_automation/app_store_step_1.png
+    :alt: App store step 1
+
+  .. image:: /_static/img/email_marketing_and_automation/app_store_step_2.png
+    :alt: App store step 2
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:22]:
```suggestion
  .. image:: /_static/img/email_marketing_and_automation/app_store_step_2.png
     :alt: App store step 2
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 7]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+      .createHmac('sha256', WEBHOOK_SECRET_KEY)
+      .update(data)
+      .digest('hex')
+    );
+
+    return crypto.timingSafeEqual(
+      Buffer.from(calculatedSignature, 'base64'),
+      Buffer.from(headerSignature, 'base64')
+    );
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:103]: I’m following indentation guidelines from RST documentation.
https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html

```suggestion
   const crypto = require('crypto');
 
   const WEBHOOK_SECRET_KEY = 'my_webhook_secret';
 
   function verifyWebhook(data, headerSignature) {
     // Calculate HMAC
     const calculatedSignature = btoa(crypto
       .createHmac('sha256', WEBHOOK_SECRET_KEY)
       .update(data)
       .digest('hex')
     );
 
     return crypto.timingSafeEqual(
       Buffer.from(calculatedSignature, 'base64'),
       Buffer.from(headerSignature, 'base64')
     );
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 8]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+Available actions
+-----------------
+
+Each action allows custom attributes to be included. You can see allowed interpolation variables by clicking `Show available variables` button:
+
+.. image:: /_static/img/email_marketing_and_automation/variables.png
+  :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:111]: One issue here is indentation, but the ALT text is not clear to me — what does "Send some payload" mean here? Should it say variables?
```suggestion
   :alt: Variables
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 9]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "campaign_id": "142",
+    "referral_status": "in_progress",
+    "advocate_email": "advocate@example.com",
+    "friend_email": "friend@example.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:128]: I’m following indentation guidelines from RST documentation.
https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html

```suggestion
   {
     "site_id": "1",
     "campaign_id": "142",
     "referral_status": "in_progress",
     "advocate_email": "advocate@example.com",
     "friend_email": "friend@example.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 10]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "email": "person-9eb230f8d189fca9@talkable-sample.com",
+    "email_optin": "true",
+    "phone_number": "+12025551111",
+    "phone_optin": "true"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:144]:
```suggestion
   {
     "email": "person-9eb230f8d189fca9@talkable-sample.com",
     "email_optin": "true",
     "phone_number": "+12025551111",
     "phone_optin": "true"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 11]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "person-40728a5b940e3247@talkable-sample.com",
+    "email_optin": "true"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:159]:
```suggestion
   {
     "site_id": "1",
     "email": "person-40728a5b940e3247@talkable-sample.com",
     "email_optin": "true"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 12]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "person-fa8e880e25bd47b3@talkable-sample.com",
+    "phone_number": "+12025551111"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:174]:
```suggestion
   {
     "site_id": "1",
     "email": "person-fa8e880e25bd47b3@talkable-sample.com",
     "phone_number": "+12025551111"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 13]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+This action automatically synchronizes all loyalty members whenever they make a loyalty action.
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "loyalty@talkable.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:188]:
```suggestion
   {
     "site_id": "1",
     "email": "loyalty@talkable.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 14]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+This action automatically synchronizes all loyalty members whenever their tier is changed.
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "loyalty@talkable.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:202]:
```suggestion
   {
     "site_id": "1",
     "email": "loyalty@talkable.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 15]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+Whenever customers unsubscribe from Talkable emails they will also become unsubscribed in your ESP.
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "person-591542c54ff21a49@talkable-sample.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:216]:
```suggestion
   {
     "site_id": "1",
     "email": "person-591542c54ff21a49@talkable-sample.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 16]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+Whenever a person shares the offer the share information will be sent to your ESP.
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "email": "person-591542c54ff21a49@talkable-sample.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:230]:
```suggestion
   {
     "site_id": "1",
     "email": "person-591542c54ff21a49@talkable-sample.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 17]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+.. code-block:: JSON
+
+  {
+    "description": "$5",
+    "amount": "5.0",
+    "coupon_code": "FR_NEW_5_OFF",
+    "reason": "click",
+    "site_id": "1"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:247]:
```suggestion
   {
     "description": "$5",
     "amount": "5.0",
     "coupon_code": "FR_NEW_5_OFF",
     "reason": "click",
     "site_id": "1"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 18]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "coupon_code": "SAMPLE-COUPON-CODE",
+    "coupon_discount_amount": "0.0"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:262]:
```suggestion
   {
     "site_id": "1",
     "coupon_code": "SAMPLE-COUPON-CODE",
     "coupon_discount_amount": "0.0"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 19]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+  {
+    "site_id": "1",
+    "type": "Event",
+    "created_at": "2024-11-25 00:00:00 -0800",
+    "event_category": "public_event",
+    "event_number": "183247241",
+    "subtotal": "100.0",
+    "currency_iso_code": "USD",
+    "email": "advocate@example.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:282]:
```suggestion
   {
     "site_id": "1",
     "type": "Event",
     "created_at": "2024-11-25 00:00:00 -0800",
     "event_category": "public_event",
     "event_number": "183247241",
     "subtotal": "100.0",
     "currency_iso_code": "USD",
     "email": "advocate@example.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 20]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+.. code-block:: JSON
+
+  {
+    "site_id": "1",
+    "campaign_id": "142",
+    "referral_status": "in_progress",
+    "advocate_email": "advocate@example.com",
+    "friend_email": "friend@example.com"
+  }
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:299]:
```suggestion
   {
     "site_id": "1",
     "campaign_id": "142",
     "referral_status": "in_progress",
     "advocate_email": "advocate@example.com",
     "friend_email": "friend@example.com"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 21]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+*Default payload:*
+
+.. code-block:: JSON
+
+  {
+    "email": "friend@example.com",
+    "site_id": "1"
+  }
+
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:316]:
```suggestion
   {
     "email": "friend@example.com",
     "site_id": "1"
   }
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 22]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+Testing
+-------
+
+Testing Custom app actions can be accomplished with the help of Webhook Tester, an external service that tests your post-receive messages.
+
+1. Visit `Webhook Tester <https://webhook.site/>`_ and click **Copy** to copy the URL you are given.
+2. Open your Custom app .
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:324]: Redundant space.
```suggestion
2. Open your Custom app.
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 23]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
 
-    function verifyWebhook(data, headerSignature) {
-        // Calculate HMAC
-        const calculatedSignature = btoa(crypto
-            .createHmac('sha256', WEBHOOK_SECRET_KEY)
-            .update(data)
-            .digest('hex')
-        );
+.. hlist::
+  :columns: 4
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:342]:
```suggestion
   :columns: 4
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 24]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
 
-        return crypto.timingSafeEqual(
-            Buffer.from(calculatedSignature, 'base64'),
-            Buffer.from(headerSignature, 'base64')
-        );
-    }
+  .. include:: /partials/talkable_ip_list.rst
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:344]:
```suggestion
   .. include:: /partials/talkable_ip_list.rst
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 25]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+Each action allows custom attributes to be included. You can see allowed interpolation variables by clicking `Show available variables` button:
+
+.. image:: /_static/img/email_marketing_and_automation/variables.png
+  :alt: Send same payload
+
+Referral Create
+...............
+
+When referral has been created this action sends you information.
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:116]:
```suggestion
Triggers when a new referral is created, sending detailed information about the referral.
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 26]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+Testing Custom app actions can be accomplished with the help of Webhook Tester, an external service that tests your post-receive messages.
+
+1. Visit `Webhook Tester <https://webhook.site/>`_ and click **Copy** to copy the URL you are given.
+2. Open your Custom app .
+3. Paste your Webhook Tester URL into **Endpoint URL** field and save.
+4. Click **Send sample payload** near the action you want to test.
+
+  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+      :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:329]: > [!IMPORTANT]  
> Please don’t forget to rename the image file.

* Fix indentation.
* Fix file name, "same payload" is a typo and does not make sense in this context.

```suggestion
  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send sample payload
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 27]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+    "event_number": "183247241",
+    "subtotal": "100.0",
+    "currency_iso_code": "USD",
+    "email": "advocate@example.com"
+  }
+
+Referral Status Change
+......................
+
+When referral has been created this action sends you information.
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:287]: This text seems to be copy-paste from "Referral Create" without a proper change.
```suggestion
This action is triggered when referral’s status changes.
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 28]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+  {
+    "site_id": "1",
+    "coupon_code": "SAMPLE-COUPON-CODE",
+    "coupon_discount_amount": "0.0"
+  }
+
+Event
+.....
+
+When Event or Purchase have been registered this actions sends you information.
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:267]:
```suggestion
Triggers when a new event or purchase is created.
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 29]
```diff
@@ -66,20 +84,261 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+
+  {
+    "site_id": "1",
+    "email": "person-591542c54ff21a49@talkable-sample.com"
+  }
+
+Offer Share
+...........
+
+Whenever a person shares the offer the share information will be sent to your ESP.
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:221]:
```suggestion
Whenever a person shares the offer, the share information will be sent to your ESP.
```

**Review by zhuravel — CHANGES_REQUESTED** [review]

**Comment by zhuravel** [comment]
@wrimer Yes and yes. Thank you!

File: source/email_marketing_and_automation/custom_app.rst [hunk 30]
```diff
@@ -51,35 +69,276 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+     );
+   }
+
+Available actions
+-----------------
+
+Each action allows custom attributes to be included. You can see allowed interpolation variables by clicking `Show available variables` button:
+
+.. image:: /_static/img/email_marketing_and_automation/variables.png
+   :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:111]: This should either say "Send **sample** payload" or "Variables" (as it refers to image other than `send_sample_payload.png`)

File: source/email_marketing_and_automation/custom_app.rst [hunk 31]
```diff
@@ -51,35 +69,276 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+.. code-block:: JSON
+
+   {
+     "email": "person-9eb230f8d189fca9@talkable-sample.com",
+     "email_optin": "true",
+     "phone_number": "+12025551111",
+     "phone_optin": "true"
+   }
+
+Sync email opt ins
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:146]: You don’t have a space in `signups` above but have in `opt ins`.
Please use either signups / optins or sign ups / opt ins consistently.

File: source/email_marketing_and_automation/custom_app.rst [hunk 32]
```diff
@@ -51,35 +69,276 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+     "type": "Event",
+     "created_at": "2024-11-25 00:00:00 -0800",
+     "event_category": "public_event",
+     "event_number": "183247241",
+     "subtotal": "100.0",
+     "currency_iso_code": "USD",
+     "email": "advocate@example.com"
+   }
+
+Referral Status Change
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:284]:
```suggestion
Referral status change
```

File: source/email_marketing_and_automation/custom_app.rst [hunk 33]
```diff
@@ -51,35 +69,276 @@ Your Talkable **Webhook security key** can be found in the Webhook set up page b
...
+*Default payload:*
+
+.. code-block:: JSON
+
+   {
+     "site_id": "1",
+     "email": "person-591542c54ff21a49@talkable-sample.com"
+   }
+
+Offer Share
```
Comment (zhuravel) [inline source/email_marketing_and_automation/custom_app.rst:218]:
```suggestion
Offer share
```

File: source/email_marketing_and_automation/hubspot.rst [hunk 34]
```diff
@@ -52,7 +52,7 @@ Please use the following guide to start using an app you need:
 
 5. Test the app by pressing on the “Send sample payload” and then check if you are seeing a test request inside your ESP:
 
-  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/hubspot.rst:56]:
```suggestion
    :alt: Send sample payload
```

File: source/email_marketing_and_automation/klaviyo.rst [hunk 35]
```diff
@@ -52,7 +52,7 @@ Please use the following guide to start using an app you need:
 
 5. Test the app by pressing on the “Send sample payload” and then check if you are seeing a test request inside your ESP:
 
-  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/klaviyo.rst:56]:
```suggestion
    :alt: Send sample payload
```

File: source/email_marketing_and_automation/mailchimp.rst [hunk 36]
```diff
@@ -52,7 +52,7 @@ Please use the following guide to start using an app you need:
 
 5. Test the app by pressing on the “Send sample payload” and then check if you are seeing a test request inside your ESP:
 
-  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/mailchimp.rst:56]:
```suggestion
    :alt: Send sample payload
```

File: source/email_marketing_and_automation/oracle_bronto.rst [hunk 37]
```diff
@@ -52,7 +52,7 @@ Please use the following guide to start using an app you need:
 
 5. Test the app by pressing on the “Send sample payload” and then check if you are seeing a test request inside your ESP:
 
-  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/oracle_bronto.rst:56]:
```suggestion
    :alt: Send sample payload
```

File: source/email_marketing_and_automation/sailthru.rst [hunk 38]
```diff
@@ -52,7 +52,7 @@ Please use the following guide to start using an app you need:
 
 5. Test the app by pressing on the “Send sample payload” and then check if you are seeing a test request inside your ESP:
 
-  .. image:: /_static/img/email_marketing_and_automation/send_same_payload.png
+  .. image:: /_static/img/email_marketing_and_automation/send_sample_payload.png
     :alt: Send same payload
```
Comment (zhuravel) [inline source/email_marketing_and_automation/sailthru.rst:56]:
```suggestion
    :alt: Send sample payload
```

**Review by zhuravel — CHANGES_REQUESTED** [review]

**Comment by wrimer** [comment]
@zhuravel chatGPT says that `sign ups/signups` and `opt-ins/optins` pairs should be used in different cases. So, i altered description with his suggestions. 
CleanShot 2024-12-03 at 22 44 34@2x
Also I've added this point for [task for our product manifest](https://talkable.atlassian.net/browse/PR-23488)
For history: Task for reviewing indentation: https://talkable.atlassian.net/browse/PR-23489

**Review by zhuravel — APPROVED** [review]

**Pull request merged**